android/src/main/java/flutter/overlay/window/flutter_overlay_window/OverlayService.java

```diff
@@ -84,24 +94,13 @@ public void onDestroy() {
     @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
-        mResources = getApplicationContext().getResources();
-        boolean isCloseWindow = intent.getBooleanExtra(INTENT_EXTRA_IS_CLOSE_WINDOW, false);
-        if (isCloseWindow) {
-            if (windowManager != null) {
-                windowManager.removeView(flutterView);
-                windowManager = null;
-                flutterView.detachFromFlutterEngine();
-                stopSelf();
-            }
-            isRunning = false;
-            return START_STICKY;
-        }
-        if (windowManager != null) {
-            windowManager.removeView(flutterView);
-            windowManager = null;
-            flutterView.detachFromFlutterEngine();
+        if (intent != null && OverlayConstants.ACTION_STOP_OVERLAY.equals(intent.getAction())) {
+            stopForeground(true);
```

Comment:
Is there any reason for detaching the service from the foreground?
Also from API 33, this method has been deprecated so you will have to pass the proper notification flag behaviour
https://developer.android.com/reference/android/app/Service#stopForeground(boolean)